pkg/operator/ceph/csi/cluster_config_test.go

```diff
@@ -19,71 +19,98 @@ package csi
+		s, err = updateCsiClusterConfig("[]", "alpha", csiClusterConfigEntry)
+		assert.NoError(t, err)
+		assert.Equal(t, s,
+			`[{"clusterID":"alpha","monitors":["1.2.3.4:5000"]}]`)
+	})
+
+	t.Run("add a 2nd mon to the current cluster", func(t *testing.T) {
+		csiClusterConfigEntry.Monitors = append(csiClusterConfigEntry.Monitors, "10.11.12.13:5000")
+		s, err = updateCsiClusterConfig(s, "alpha", csiClusterConfigEntry)
+		assert.NoError(t, err)
+		cc, err := parseCsiClusterConfig(s)
+		assert.NoError(t, err)
+		assert.Equal(t, len(cc), 1)
+		assert.Equal(t, cc[0].ClusterID, "alpha")
+		assert.Contains(t, cc[0].Monitors, "1.2.3.4:5000")
+		assert.Contains(t, cc[0].Monitors, "10.11.12.13:5000")
+		assert.Equal(t, len(cc[0].Monitors), 2)
+	})
+
+	t.Run("add a 2nd cluster with 3 mons", func(t *testing.T) {
+		s, err = updateCsiClusterConfig(s, "beta", csiClusterConfigEntry2)
+		assert.NoError(t, err)
+		cc, err := parseCsiClusterConfig(s)
+		assert.NoError(t, err)
+		assert.Equal(t, len(cc), 2)
+		assert.Equal(t, cc[0].ClusterID, "alpha")
+		assert.Contains(t, cc[0].Monitors, "1.2.3.4:5000")
+		assert.Contains(t, cc[0].Monitors, "10.11.12.13:5000")
+		assert.Equal(t, len(cc[0].Monitors), 2)
+		assert.Equal(t, cc[1].ClusterID, "beta")
+		assert.Contains(t, cc[1].Monitors, "20.1.1.1:5000")
+		assert.Contains(t, cc[1].Monitors, "20.1.1.2:5000")
+		assert.Contains(t, cc[1].Monitors, "20.1.1.3:5000")
+		assert.Equal(t, len(cc[1].Monitors), 3)
+	})
+
+	t.Run("remove a mon from the 2nd cluster", func(t *testing.T) {
+		i := 2
+		// Remove last element of the slice
+		csiClusterConfigEntry2.Monitors = append(csiClusterConfigEntry2.Monitors[:i], csiClusterConfigEntry2.Monitors[i+1:]...)
+		s, err = updateCsiClusterConfig(s, "beta", csiClusterConfigEntry2)
+		assert.NoError(t, err)
+		cc, err := parseCsiClusterConfig(s)
+		assert.NoError(t, err)
+		assert.Equal(t, len(cc), 2)
+		assert.Equal(t, cc[0].ClusterID, "alpha")
+		assert.Contains(t, cc[0].Monitors, "1.2.3.4:5000")
+		assert.Contains(t, cc[0].Monitors, "10.11.12.13:5000")
+		assert.Equal(t, len(cc[0].Monitors), 2)
+		assert.Equal(t, cc[1].ClusterID, "beta")
+		assert.Contains(t, cc[1].Monitors, "20.1.1.1:5000")
+		assert.Contains(t, cc[1].Monitors, "20.1.1.2:5000")
+		assert.Equal(t, len(cc[1].Monitors), 2)
+	})
 
-	// remove a mon from the 2nd cluster
-	delete(mons2, "blam")
-	s, err = updateCsiClusterConfig(s, "beta", mons2)
-	assert.NoError(t, err)
-	cc, err = parseCsiClusterConfig(s)
-	assert.NoError(t, err)
-	assert.Equal(t, len(cc), 2)
-	assert.Equal(t, cc[0].ClusterID, "alpha")
-	assert.Contains(t, cc[0].Monitors, "1.2.3.4:5000")
-	assert.Contains(t, cc[0].Monitors, "10.11.12.13:5000")
-	assert.Equal(t, len(cc[0].Monitors), 2)
-	assert.Equal(t, cc[1].ClusterID, "beta")
-	assert.Contains(t, cc[1].Monitors, "20.1.1.1:5000")
-	assert.Contains(t, cc[1].Monitors, "20.1.1.2:5000")
-	assert.Equal(t, len(cc[1].Monitors), 2)
+	t.Run("add a 3rd cluster with subvolumegroup", func(t *testing.T) {
```

Comment:
Is there a unit test that shows the subvolume groups are preserved if the mons are updated?

Reply:
Added